app/main.py

```diff
@@ -1,3 +1,44 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km):
+        self.km = km
+
+    def __str__(self):
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self):
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other):
+        if isinstance(other, (int, float)):
+            self.km = self.km + other
+        else:
+            self.km = self.km + other.km
+        return self
+
+    def __iadd__(self, other):
+        if isinstance(other, (int, float)):
```

Comment:
Since you're using this check several times across your code, you should implement a new method that checks if other is a number or `Distance` object, and returns `other` or `other.km` respectively.